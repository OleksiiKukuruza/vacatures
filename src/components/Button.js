import PropTypes from 'prop-types';
import styled from 'styled-components';

const Button = styled.button`
  cursor: pointer;
  width: ${props => props.fullWidth ? '100%' : 'auto'};
  font-size: 1.3rem;
  background-color: ${props => props.theme.primary.contrastText};
  color: ${props => props.theme.primary.main};
  border: none;
  outline: none;
  border-radius: ${props => props.theme.spacingUnit / 2}px;
  padding: ${props => props.theme.spacingUnit}px 0;
  transition: filter 300ms ease-in;
  
  :hover {
    filter: brightness(85%);
  }
`;

Button.propTypes = {
  fullWidth: PropTypes.bool
};

export default Button;
