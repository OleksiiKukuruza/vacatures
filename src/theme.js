const theme = {
  primary: {
    main: '#54b6c8',
    contrastText: '#fff'
  },
  background: '#f1f4f5',
  spacingUnit: 8
};

export default theme;
