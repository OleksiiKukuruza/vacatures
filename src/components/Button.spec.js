import React from 'react';
import { shallow } from 'enzyme';
import theme from '../theme';
import Button from './Button';

const setup = (Component, props) => {
  const wrapper = shallow(<Component {...props} />);

  return {
    wrapper
  };
};

describe('Button', () => {
  it('renders default width without crashing', () => {
    const { wrapper } = setup(Button, {theme});
    expect(wrapper).toMatchSnapshot();
  });

  it('renders full width without crashing', () => {
    const { wrapper } = setup(Button, {theme, fullWidth: true});
    expect(wrapper).toMatchSnapshot();
  });
});
