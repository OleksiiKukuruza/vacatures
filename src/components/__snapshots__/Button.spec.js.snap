// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders default width without crashing 1`] = `
.c0 {
  cursor: pointer;
  width: auto;
  font-size: 1.3rem;
  background-color: #fff;
  color: #54b6c8;
  border: none;
  outline: none;
  border-radius: 4px;
  padding: 8px 0;
  -webkit-transition: filter 300ms ease-in;
  transition: filter 300ms ease-in;
}

.c0:hover {
  -webkit-filter: brightness(85%);
  filter: brightness(85%);
}

<button
  className="c0"
/>
`;

exports[`Button renders full width without crashing 1`] = `
.c0 {
  cursor: pointer;
  width: 100%;
  font-size: 1.3rem;
  background-color: #fff;
  color: #54b6c8;
  border: none;
  outline: none;
  border-radius: 4px;
  padding: 8px 0;
  -webkit-transition: filter 300ms ease-in;
  transition: filter 300ms ease-in;
}

.c0:hover {
  -webkit-filter: brightness(85%);
  filter: brightness(85%);
}

<button
  className="c0"
/>
`;
